Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1642,15 +1648,27 @@ open class NavigationMapView: UIView {
         }
     }
     
-    private func waypoints(on routes: [Route], closeTo point: CGPoint) -> [Waypoint]? {
-        let tapCoordinate = mapView.mapboxMap.coordinate(for: point)
-        let multipointRoutes = routes.filter { $0.legs.count > 1}
+    /**
+     Returns a list of waypoints, that are located on the routes with more than one leg and are
+     close to a certain point and are within threshold distance defined in
+     `NavigationMapView.tapGestureDistanceThreshold`.
+     
+     - parameter routes: List of the routes.
+     - parameter point: Point on the screen.
+     - returns: List of the waypoints, which were found. If no routes have more than one leg, `nil`
+     will be returned.
+     */
+    public func waypoints(on routes: [Route], closeTo point: CGPoint) -> [Waypoint]? {
+        // In case if route does not contain more than one leg - do nothing.
+        let multipointRoutes = routes.filter({ $0.legs.count > 1 })
```

Comment:
Is this method only for intermediate waypoints? That isn’t especially clear from the method name, since the route’s origin and destination are also known as waypoints, and so are silent waypoints. We could rename the method `legSeparatingWaypoints(on:closeTo:)` to make that clearer perhaps.